@jjcollinge you alright to take another look at this now? The build will fail as it's be reconfigured for #9 to use storage account url but have tested and int test are good

```
bash -f ./scripts/run_integration_tests.sh
=== RUN   TestLockingEnd2End_Simple
--- PASS: TestLockingEnd2End_Simple (0.94s)
	locking_integration_test.go:40: Acquired lease: 3c46710b-d628-404e-8df5-95dbe1e96de4
=== RUN   TestLockingEnd2End_UnlockOrRenewWithoutLocking
--- PASS: TestLockingEnd2End_UnlockOrRenewWithoutLocking (0.19s)
	locking_integration_test.go:96: Error returned as expected: Lock not acquired, can't unlock
	locking_integration_test.go:103: Error returned as expected: Lock not acquired, can't renew
=== RUN   TestLockingEnd2End_Defaults
--- PASS: TestLockingEnd2End_Defaults (0.86s)
	locking_integration_test.go:132: Acquired lease: c0c11021-1b61-4242-8943-fcc156f0903a
=== RUN   TestLockingEnd2End_AutoRenewal
--- PASS: TestLockingEnd2End_AutoRenewal (20.51s)
	locking_integration_test.go:181: Acquired lease: da8a5853-d156-4fbf-a451-85dec3da37f9
=== RUN   TestLockingEnd2End_InvalidLockName
--- PASS: TestLockingEnd2End_InvalidLockName (0.00s)
	locking_integration_test.go:221: lock name: lock1~~~@@@ must be alphanumberic with no characters other than '-' (regex '^[a-z0-9]+(-[a-z0-9]+)*$')
=== RUN   TestLockingEnd2End_UseLockTwice
--- PASS: TestLockingEnd2End_UseLockTwice (0.50s)
	locking_integration_test.go:247: Acquired lease: 14d82c75-a516-4de6-bd82-0fe442a37c65
=== RUN   TestLockingEnd2End_UseUnlockTwice
--- PASS: TestLockingEnd2End_UseUnlockTwice (0.38s)
	locking_integration_test.go:283: Acquired lease: 200040d1-bc72-4518-b06d-bece38ffc366
=== RUN   TestLockingEnd2End_LockRetry
--- PASS: TestLockingEnd2End_LockRetry (18.68s)
	locking_integration_test.go:323: Acquired lease: c9e97403-bc84-4f55-bdbb-682d4ba323c9
=== RUN   TestAutoRenewLockBehavior_Normal
--- PASS: TestAutoRenewLockBehavior_Normal (6.00s)
=== RUN   TestAutoRenewLockBehavior_Fail
--- PASS: TestAutoRenewLockBehavior_Fail (0.50s)
	locking_test.go:57: Lock lost signaled as expected
=== RUN   TestRetryObtainingLockBehavior
--- PASS: TestRetryObtainingLockBehavior (3.56s)
	locking_test.go:98: Lock obtained after 4 attempts
=== RUN   TestPanicOnLostLock
--- PASS: TestPanicOnLostLock (0.01s)
=== RUN   TestIsValidLockName
=== RUN   TestIsValidLockName/ValidLockName1
=== RUN   TestIsValidLockName/ValidLockName2_toLowerCaps
=== RUN   TestIsValidLockName/InvalidLockName_Long
=== RUN   TestIsValidLockName/InvalidLockName_@
--- PASS: TestIsValidLockName (0.00s)
    --- PASS: TestIsValidLockName/ValidLockName1 (0.00s)
    --- PASS: TestIsValidLockName/ValidLockName2_toLowerCaps (0.00s)
    --- PASS: TestIsValidLockName/InvalidLockName_Long (0.00s)
    --- PASS: TestIsValidLockName/InvalidLockName_@ (0.00s)
PASS
ok  	github.com/lawrencegripper/goazurelocking	52.137s
```

--- a/README.md
+++ b/README.md
@@ -39,6 +39,9 @@ if err != nil {
    
 ```
 
+> If you look at the storage account your using you'll see a container called `azlockcontainer` under it will be empty blobs for each of the locks you have created. You can 
+	configure [Azure Storage Lifecycle Management](https://azure.microsoft.com/en-us/blog/azure-blob-storage-lifecycle-management-public-preview/) to clear these down periodically
+
 
 ## Advanced
 

